solo-chains/node/tanssi-relay-service/src/lib.rs

```diff
@@ -14,6 +14,602 @@
+        // Inherent isn't checked and can not return any error
+        None
+    }
+}
+
+/// Creates new development full node with manual seal
+pub fn build_full<OverseerGenerator: OverseerGen>(
+    sealing: Sealing,
+    config: Configuration,
+    mut params: NewFullParams<OverseerGenerator>,
+) -> Result<NewFull, Error> {
+    let is_polkadot = config.chain_spec.is_polkadot();
+
+    params.overseer_message_channel_capacity_override = params
+        .overseer_message_channel_capacity_override
+        .map(move |capacity| {
+            if is_polkadot {
+                gum::warn!("Channel capacity should _never_ be tampered with on polkadot!");
+            }
+            capacity
+        });
+
+    match config.network.network_backend {
+        sc_network::config::NetworkBackendType::Libp2p => {
+            new_full::<_, sc_network::NetworkWorker<Block, Hash>>(sealing, config, params)
+        }
+        sc_network::config::NetworkBackendType::Litep2p => {
+            new_full::<_, sc_network::Litep2pNetworkBackend>(sealing, config, params)
+        }
+    }
+}
+
+fn new_full<
+    OverseerGenerator: OverseerGen,
+    Network: sc_network::NetworkBackend<Block, <Block as BlockT>::Hash>,
+>(
+    sealing: Sealing,
+    mut config: Configuration,
+    NewFullParams {
+        telemetry_worker_handle,
+        ..
+    }: NewFullParams<OverseerGenerator>,
+) -> Result<NewFull, Error> {
+    let role = config.role.clone();
+
+    let basics = new_partial_basics(&mut config, telemetry_worker_handle)?;
+
+    let prometheus_registry = config.prometheus_registry().cloned();
+
+    let keystore = basics.keystore_container.local_keystore();
+
+    let select_chain = SelectRelayChain::new_longest_chain(basics.backend.clone());
+
+    let service::PartialComponents::<_, _, SelectRelayChain<_>, _, _, _> {
+        client,
+        backend,
+        mut task_manager,
+        keystore_container,
+        select_chain,
+        import_queue,
+        transaction_pool,
+        other: (block_import, babe_link, slot_duration, mut telemetry),
+    } = new_partial::<SelectRelayChain<_>>(&mut config, basics, select_chain)?;
+
+    let metrics = Network::register_notification_metrics(
+        config.prometheus_config.as_ref().map(|cfg| &cfg.registry),
+    );
+
+    let net_config =
+        sc_network::config::FullNetworkConfiguration::<_, _, Network>::new(&config.network);
+
+    let (network, system_rpc_tx, tx_handler_controller, network_starter, sync_service) =
+        service::build_network(service::BuildNetworkParams {
+            config: &config,
+            net_config,
+            client: client.clone(),
+            transaction_pool: transaction_pool.clone(),
+            spawn_handle: task_manager.spawn_handle(),
+            import_queue,
+            block_announce_validator_builder: None,
+            warp_sync_params: None,
+            block_relay: None,
+            metrics,
+        })?;
+
+    if config.offchain_worker.enabled {
```

Comment:
I dont think we need `offchain_worker` for dev service right?

Reply:
Offchain workers can be used by some pallets including Babe. So, better if we can enable it just so that we can fully test any pallet that uses off chain workers.

Reply:
alright, thanks!